### PR TITLE
Add changelog/README.rst and streamline our PR template text

### PR DESCRIPTION
**Pull request by nicoddemus**
This streamlines the PR template text and adds a more in-depth explanation
about how the changelog entries work because this topic is a common source of
confusion:

- How to name the files.
- Which formatting to use (people in general assume it is Markdown).
- Recommend adding `.rst` extension to changelog files to help with the
  above (`towncrier` doesn't care).

This was heavily inspired by the excellent `python-trio/trio` docs, I hope @njsmith doesn't mind. 😁

### CODE REVIEW

**Comment by njsmith** [comment]
Glad to help :+1:

File: .github/PULL_REQUEST_TEMPLATE.md [hunk 1]
```diff
@@ -1,15 +1,14 @@
...
+just a guideline):
 
-- [ ] Add a new news fragment into the changelog folder
-  * name it `$issue_id.$type` for example (588.bugfix)
-  * if you don't have an issue_id change it to the pr id after creating the pr
-  * ensure type is one of `removal`, `feature`, `bugfix`, `vendor`, `doc` or `trivial`
-  * Make sure to use full sentences with correct case and punctuation, for example: "Fix issue with non-ascii contents in doctest text files."
-- [ ] Target: for `bugfix`, `vendor`, `doc` or `trivial` fixes, target `master`; for removals or features target `features`;
-- [ ] Make sure to include reasonable tests for your change if necessary
+- [ ] Create a new changelog file into the `changelog` folder, with a name like `<ISSUE NUMBER>.<TYPE>.rst`. See [changelog/README.rst](/changelog/README.rst) for details.
```
Comment (The-Compiler) [inline .github/PULL_REQUEST_TEMPLATE.md]: into -> in

File: changelog/README.rst [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+
+Make sure to use full sentences with correct case and punctuation, for example: *Fix issue with non-ascii contents in doctest text files.*
+
+Each file should be named like ``<ISSUE>.<TYPE>.rst``, where
+``<ISSUE>`` is an issue number, and ``<TYPE>`` is one of:
+
+* ``feature``: new user facing features, like new command-line options and new behavior.
+* ``bugfix``: fixes a reported bug.
+* ``doc``: documentation improvement, like rewording an entire session or adding missing docs.
+* ``removal``: feature deprecation or removal;
```
Comment (The-Compiler) [inline changelog/README.rst]: It looks like you switched from `.` to `;` here?

File: changelog/README.rst [hunk 3]
```diff
@@ -0,0 +1,30 @@
...
+* ``vendor``: vendoring news;
```
Comment (The-Compiler) [inline changelog/README.rst]: Might want to be a bit verbose here and give some context on what vendoring means - something like "Changes in packages vendored in pytest." or so? On the other hand, didn't we get rid of vendoring anyways?

Comment (nicoddemus) [reply changelog/README.rst]: Good point, updated.

We are not vendoring anything more at the moment, but we might do it again in the future. I guess we can always remove this later if we officially abolish the practice.

File: changelog/README.rst (hunk 2 above)
Comment (nicoddemus) [reply changelog/README.rst]: Done, thanks!

File: .github/PULL_REQUEST_TEMPLATE.md (hunk 1 above)
Comment (nicoddemus) [reply .github/PULL_REQUEST_TEMPLATE.md]: Done thanks!

**Comment by nicoddemus** [comment]
Applied all your suggestions @The-Compiler, thanks!

**Comment by The-Compiler** [comment]
LGTM now, thanks!

**Comment by The-Compiler** [comment]
Huh, tried to merge this from my phone but I get a 404 - feel free to merge it!

**Comment by The-Compiler** [comment]
Back on my laptop, let's merge! 🎉

**Pull request merged**